src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -93,9 +93,10 @@ public FeatureManagementKeyValueAdapter(FeatureFilterTracing featureFilterTracin
 
                     keyValues.Add(new KeyValuePair<string, string>($"{variantsPath}:{FeatureManagementConstants.Name}", featureVariant.Name));
 
-                    if (featureVariant.ConfigurationValue != null)
+                    foreach (KeyValuePair<string, string> kvp in new JsonFlattener().FlattenJson(featureVariant.ConfigurationValue))
                     {
-                        keyValues.Add(new KeyValuePair<string, string>($"{variantsPath}:{FeatureManagementConstants.ConfigurationValue}", featureVariant.ConfigurationValue));
+                        keyValues.Add(new KeyValuePair<string, string>($"{variantsPath}:{FeatureManagementConstants.ConfigurationValue}" + 
+                            (string.IsNullOrEmpty(kvp.Key) ? "" : $":{kvp.Key}"), kvp.Value));
```

Comment:
Does this logic work with a primative value? It will as long as 
```
new JsonFlattener().FlattenJson(featureVariant.ConfigurationValue)
```
returns a kvp of `"": primative`, but that isn't obvious to me.

Reply:
In the `WithVariants` unit test it does work with string, I tested with int and bool as well but didn't add it to the test. I can add it to be thorough.

Reply:
Oh, I didn't see the test. Yeah adding the last option might not be a bad idea.